rfcs/0103-flake-tags-metadata.md

```diff
@@ -0,0 +1,150 @@
+---
+feature: Flakes metadata (tags)
+start-date: 2021-09-02
+author: Bryan Honof
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Having a bit more information about what a flake provides could be useful for
+identifying, and categorising, the software the flake contains. This way
+existing tools like search.nixos.org can use this information for
+indexing/searching.
+
+# Motivation
+[motivation]: #motivation
+
+Just as NixPkgs uses a file structure to categorise software, e.g.
+`./pkgs/games/minetest`, tags in a flake could function in the same way. Having
+a tag called "games" in a flake that provides the minetest package as example,
+will indicate that this flake possibly provides a package that happens to be a
+game.
+
+As mentioned in the summary, one of the use cases could be that of
+search.nixos.org. It will also give the user an opportunity to specify what they
+think the software's category is. Another use case could be that of NixOps.
+Currently, NixOps doesn't use flakes, but shouldn't be too difficult to convert
+to one. Once that has happened, a NixOps plugin could identify itself to be a
+plugin by providing the tags `plugin` and `nixops`, inside their flake.
+
+The expected outcome is that flakes will be better identifyable through a simple
+interface.
+
+# Detailed design
+[design]: #detailed-design
+
+The `tags` attribute in a flake should NOT be mandatory, but optional. It is
```

Comment:
since flakes are heavily coupled to git, and `tag` is already a well-defined term when referring to version control, maybe the use of `label` would be more appropriate?

Otherwise I could see a scenario where someone says, "I cut a [git] tag for the product, make sure to update your [flake-referenced git] tags in your flakes. I also added another [flake] tag, to help with discoverability"

Reply:
That's a good comment, I hadn't thought about this myself. I'll make the edit when I get back behind my PC.